torch/_ops.py

```diff
@@ -782,6 +775,85 @@ def tags(self):
     # TODO: add more methods to expose information about input and output arguments
 
 
+# TorchBindOpOverload are those custom ops which have at least one overload's
+# schema consists of torch.ScriptObject (i.e. custom class) input.
+# TorchBindOpOverload will skip C++ dispatcher and purely dispatched in python
+# when its inputs contain FakeScriptObject in a similar way as higher order ops.
+class TorchBindOpOverload(OpOverload):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _fallthrough_keys(self) -> List[DispatchKey]:
+        _DEFAULT_FALLTHROUGH_KEYS = [
+            DispatchKey.Autograd,
+            DispatchKey.AutogradCPU,
+            DispatchKey.AutogradCUDA,
+            DispatchKey.ADInplaceOrView,
+            DispatchKey.PythonTLSSnapshot,
+        ]
+        return [
+            key
+            for key in _DEFAULT_FALLTHROUGH_KEYS
+            if key not in self.py_kernels
+            or self.py_kernels[key] is torch.library.fallthrough_kernel
+        ]
+
+    # use `self_` to avoid naming collide with arguments that
+    # are named "self". This way, they can be called by kwargs.
+    def __call__(self_, *args, **kwargs):  # noqa: B902
+        if _must_dispatch_in_python(args, kwargs):
+            # When any inputs are FakeScriptObject, we need to
+            # skip c++ dispatcher and dispatch in python through _get_dispatch of python_dispatcher.
+            return self_._dispatch_in_python(args, kwargs, self_._fallthrough_keys())
```

Comment:
Compute _fallthrough_keys on the fly makes sure we don't add kernels to the dispatching logic, so that when tracing finishes, the dispatch key implementation of ops remain the same.